test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/lang/CharacterUnicodeBlockTest.scala

```diff
@@ -0,0 +1,895 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.testsuite.javalib.lang
+
+import java.lang.Character.UnicodeBlock
+import org.junit.Test
+import org.junit.Assert._
+import org.scalajs.testsuite.utils.AssertThrows._
+
+class CharacterUnicodeBlockTest {
+
+  @Test def forName_Normalizations(): Unit = {
+    assertEquals(UnicodeBlock.BASIC_LATIN, UnicodeBlock.forName("Basic Latin"))
+    assertEquals(UnicodeBlock.BASIC_LATIN, UnicodeBlock.forName("Basic LatiN"))
+    assertEquals(UnicodeBlock.BASIC_LATIN, UnicodeBlock.forName("BASIC_LATIN"))
+    assertEquals(UnicodeBlock.BASIC_LATIN, UnicodeBlock.forName("BASIC_LATIN"))
+
+    assertThrows(classOf[IllegalArgumentException], UnicodeBlock.forName("Basic Lat iN"))
+    assertThrows(classOf[IllegalArgumentException], UnicodeBlock.forName("Basic Lat in"))
+    assertThrows(classOf[IllegalArgumentException], UnicodeBlock.forName("Basic-Latin"))
+
+    assertEquals(UnicodeBlock.ARABIC_EXTENDED_A, UnicodeBlock.forName("Arabic Extended-A"))
+    assertEquals(UnicodeBlock.ARABIC_EXTENDED_A, UnicodeBlock.forName("ARABIC EXTENDED-A"))
+    assertEquals(UnicodeBlock.ARABIC_EXTENDED_A, UnicodeBlock.forName("Arabic_Extended_A"))
+    assertEquals(UnicodeBlock.ARABIC_EXTENDED_A, UnicodeBlock.forName("ARABIC_EXTENDED_A"))
+    assertThrows(classOf[IllegalArgumentException], UnicodeBlock.forName("Arabic Extended_A"))
+    assertThrows(classOf[IllegalArgumentException], UnicodeBlock.forName("Arabic_Extended-A"))
+    assertThrows(classOf[IllegalArgumentException], UnicodeBlock.forName("Arabic_ExtendedA"))
+    assertEquals(UnicodeBlock.ARABIC_EXTENDED_A, UnicodeBlock.forName("ArabicExtended-A"))
+    assertEquals(UnicodeBlock.ARABIC_EXTENDED_A, UnicodeBlock.forName("ARABICEXTENDED-A"))
+    assertThrows(classOf[IllegalArgumentException], UnicodeBlock.forName("Arabic ExtendedA"))
+    assertThrows(classOf[IllegalArgumentException], UnicodeBlock.forName("Arabic_ExtendedA"))
+    assertThrows(classOf[IllegalArgumentException], UnicodeBlock.forName("ArabicExtendedA"))
+
+    assertEquals(UnicodeBlock.CJK_UNIFIED_IDEOGRAPHS_EXTENSION_A,
+        UnicodeBlock.forName("CJK Unified Ideographs Extension A"))
+    assertThrows(classOf[IllegalArgumentException],
+        UnicodeBlock.forName("CJK Unified Ideographs Extension-A"))
+  }
+
+  @Test def of_Char(): Unit = {
+    assertEquals(UnicodeBlock.BASIC_LATIN, UnicodeBlock.of(0x0.toChar))
```

Comment:
Were those tests generated to be exhaustive? If yes, we should include the script that was used to generate them as well, since we'll have to redo it one day when we use a more recent version of the JDK/Unicode.

Reply:
I think I  got them from harmony, I'll look into a code-gen solution